frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditComponentIdRow/EditComponentIdRow.test.tsx

```diff
@@ -107,4 +107,19 @@ describe('EditComponentIdRow', () => {
       screen.getByText(textMock('ux_editor.modal_properties_component_id_not_unique_error')),
     ).toBeInTheDocument();
   });
+
+  it('should render css-class when we have duplicatedId', async () => {
+    jest.spyOn(console, 'error').mockImplementation();
+    jest.doMock('../../../../utils/formLayoutUtils', () => ({
+      ...jest.requireActual('../../../../utils/formLayoutUtils'),
+      findLayoutsContainingDuplicateComponents: jest.fn().mockReturnValue({
+        duplicateComponents: ['test'],
+        layoutName: 'test',
+      }),
+    }));
+    require('./EditComponentIdRow');
+    await studioRender();
+    const container = screen.getByTestId('edit-component-id-row-container');
+    expect(container).toMatchSnapshot();
```

Comment:
I'm not sure if this test currently verifies anything, as all CSS modules are mocked to `undefined` in our Jest config. You can confirm this by looking at the snapshot file, where the CSS class is missing from the `container` div.

I think we can solve this neatly by mocking the class names to the css module name like this.

Something like this:
Changing [frontend/testing/mocks/styleMock.js](https://github.com/Altinn/altinn-studio/blob/3d23b754dee58b05599b2a40e62ade76892298cc/frontend/testing/mocks/styleMock.js#L1) to this([ref.](https://jestjs.io/docs/webpack#mocking-css-modules)):
```js
module.exports = new Proxy({}, {
  get: function getter(target, key) {
    if (key === '__esModule') {
      return false;
    }
    return key;
  }
});
```
Should let you test if the class is applied with 
```js
expect(container).toHaveClass(classes.duplicatedIdField);
```

This will likely make other tests using snapshots fail, but from my searching I only found two. Both seem very vague about what they are actually testing, so it might be acceptable to remove them 🤔 Here is a link to the tests i found: [this](https://github.com/Altinn/altinn-studio/blob/3d23b754dee58b05599b2a40e62ade76892298cc/frontend/packages/shared/src/navigation/main-header/ProfileMenu/ProfileMenu.test.tsx#L20) and the one below it

What do you think?

Reply:
yes, I  agree 😊 thanks for your feedback , the test actually  did not work , i had a discusion about it with David, and we would go through it again. but I will  try your sugisstion first 😊

Reply:
Agreed 👍